File: test/data/cpp/test01/ConsoleOutControllee.cpp
#include "ConsoleOutControllee.h"
#include <stdio.h>

ConsoleOutControllee::ConsoleOutControllee()
{

}
ConsoleOutControllee::~ConsoleOutControllee()
{

}
void ConsoleOutControllee::DoAction1()
{
    printf("Action1");
}
void ConsoleOutControllee::DoAction2()
{
    printf("Action2");
}
void ConsoleOutControllee::DoAction3()
{
    printf("Action3");
}
void ConsoleOutControllee::NoTransition(const char* state, const char* transition)
{
    printf("NoTransition(%s: %s)", state, transition);
}

void ConsoleOutControllee::OverTransition(const char* transition)
{
    printf("OverTransition(%s)", transition);
}

File: test/data/cs/test01/ConsoleOutControllee.cs
public class ConsoleOutControllee : IControllee
{
    public void DoAction1()
    {
        Console.WriteLine("Action1");
    }
    public void DoAction2()
    {
        Console.WriteLine("Action2");
    }
    public void DoAction3()
    {
        Console.WriteLine("Action3");
    }
    public void NoTransition(string state, string transition)
    {
        Console.WriteLine($"NoTransition({state}: {transition})");
    }
    public void OverTransition(string transition)
    {
        Console.WriteLine($"OverTransition({transition})");
    }
}